larreco/RecoAlg/TrackMomentumCalculator.cxx

```diff
@@ -120,6 +183,103 @@ namespace {
     std::vector<double> const xmeas_;
     std::vector<double> const ymeas_;
     std::vector<double> const eymeas_;
+    double const correction_;
+  };
+
+  class FcnWrapperLLHD {
+  public:
+    explicit FcnWrapperLLHD(std::vector<double>& dEi,
+                            std::vector<double>& dEj,
+                            std::vector<double>& dthij,
+                            std::vector<double>& ind,
+                            std::vector<bool>& dthij_valid,
+                            double stepsize,
+                            double correction)
+      : dEi_{dEi}
+      , dEj_{dEj}
+      , dthij_{dthij}
+      , ind_{ind}
+      , dthij_valid_{dthij_valid}
+      , stepsize_{stepsize}
+      , correction_{correction}
+    {}
+
+    double my_mcs_llhd(double const* x) const
+    {
+      double result = 0.0;
+
+      double red_length = stepsize_ / rad_length;
+
+      double p = x[0];
+      double theta0 = x[1];
+
+      // Total initial energy of the muon (converting the input "p" into energy with muon mass)
+      double Etot = std::sqrt(cet::sum_of_squares(p, m_muon));
```

Comment:
Suggest using `std::hypot(p, m_muon)` here and anywhere else `std::sqrt(cet::sum_of_squares(...))` is used.